File: python/customer/update_customer.py
from ultracart.apis import CustomerApi
from ultracart.api_client import ApiException

from samples import api_client
from customer_functions import insert_sample_customer, delete_sample_customer

def update_customer_example():
    """
    Demonstrate customer retrieval, update, and deletion:
    1. Insert a sample customer
    2. Retrieve customer details
    3. Update billing address
    4. Verify the update
    5. Delete the sample customer
    """
    try:
        # Insert a sample customer
        customer_oid = insert_sample_customer()

        # Create API client
        customer_api = CustomerApi(api_client())

        # Specify expansion fields (just want address fields)
        # See https://www.ultracart.com/api/#resource_customer.html for all expansion values
        expand = "billing,shipping"

        # Retrieve customer details
        customer = customer_api.get_customer(customer_oid, expand).customer

        # TODO: Modify customer details
        # Change billing address (assuming first billing entry)
        customer.billing[0].address2 = 'Apartment 101'

        # Update customer
        # Notice expand is passed to update to get back the same fields for comparison
        api_response = customer_api.update_customer(customer_oid, customer, expand)

        # Verify the update
        print(api_response.customer)

        # Clean up the sample customer
        delete_sample_customer(customer_oid)

    except ApiException as e:
        print('An ApiException occurred. Please review the following error:')
        print(e)
        raise

# Run the function if the script is executed directly
if __name__ == '__main__':
    update_customer_example()

File: typescript/customer/UpdateCustomer.ts
import {Customer, CustomerResponse} from 'ultracart_rest_api_v2_typescript';
import {customerApi} from '../api';
import {CustomerFunctions} from './CustomerFunctions';


export class UpdateCustomer {
    /**
     * Executes a customer update workflow
     * Inserts a sample customer, updates their billing address,
     * and then deletes the sample customer
     */
    public static async Execute(): Promise<void> {
        try {
            // Insert a sample customer and get their OID
            const customerOid: number = await CustomerFunctions.insertSampleCustomer();

            // just want address fields. see https://www.ultracart.com/api/#resource_customer.html for all expansion values
            const expand: string = "billing,shipping";

            // Retrieve the customer
            const customerResponse = await customerApi.getCustomer({customerProfileOid: customerOid, expand});
            const customer: Customer | undefined = customerResponse.customer;

            if (customer === undefined) {
                console.error("getCustomer returned undefined, cannot update.");
                process.exit(1);
            }

            // TODO: do some edits to the customer. Here we will change some billing fields.
            if (customer.billing && customer.billing.length > 0) {
                customer.billing[0].address2 = "Apartment 101";
            }

            // notice expand is passed to update as well since it returns back an updated customer object.
            // we use the same expansion, so we get back the same fields and can do comparisons.
            const apiResponse: CustomerResponse = await customerApi.updateCustomer({
                customerProfileOid: customerOid,
                customer: customer,
                expand: expand
            });

            // verify the update
            console.log(apiResponse.customer);

            // Delete the sample customer
            await CustomerFunctions.deleteSampleCustomer(customerOid);
        } catch (e) {
            console.error("An unexpected error occurred:", e);
            process.exit(1);
        }
    }
}